cmake/setup_cached_variables.cmake

```diff
@@ -345,6 +346,12 @@ OPTION(DEAL_II_DOXYGEN_USE_MATHJAX
   )
 MARK_AS_ADVANCED(DEAL_II_DOXYGEN_USE_MATHJAX)
 
+OPTION(DEAL_II_COMPILE_EXAMPLES
+  "If set to ON, all configurable example executables will be built and installed as well. If set to OFF, the examples component only installs the source code of example steps."
+  OFF
```

Comment:
I thought we had agreed that doing so is a useful thing. Maybe make the default `ON`?